hydra-cluster/src/Hydra/Cluster/Scenarios.hs

```diff
@@ -302,150 +295,109 @@ singlePartyOpenAHead tracer workDir node hydraScriptsTxId callback =
       send n1 $ input "Init" []
       headId <- waitMatch (10 * blockTime) n1 $ headIsInitializingWith (Set.fromList [alice])
 
-      -- Prepare a script output on the network
-      let script = fromPlutusScript @PlutusScriptV2 $ Plutus.alwaysSucceedingNAryFunction 2
-          scriptAddress = mkScriptAddress @PlutusScriptV2 networkId script
-          reedemer = 1 :: Integer
-          datum = 2 :: Integer
-          scriptInfo = ScriptInfo (toScriptData reedemer) (Just $ toScriptData datum) script
-      (scriptTxIn, scriptTxOut) <- createOutputAtAddress node scriptAddress (mkTxOutDatumHash datum)
-
-      -- Commit the script output using known witness
-      let clientPayload =
-            DraftCommitTxRequest
-              { utxoToCommit =
-                  UTxO.singleton
-                    ( scriptTxIn
-                    , TxOutWithWitness
-                        { txOut = scriptTxOut
-                        , witness = Just scriptInfo
-                        }
-                    )
-              }
-      res <-
-        runReq defaultHttpConfig $
-          req
-            POST
-            (http "127.0.0.1" /: "commit")
-            (ReqBodyJson clientPayload)
-            (Proxy :: Proxy (JsonResponse DraftCommitTxResponse))
-            (port $ 4000 + hydraNodeId)
-
-      let DraftCommitTxResponse{commitTx} = responseBody res
-      submitTx node commitTx
-
-      lockedUTxO <- waitMatch (10 * blockTime) n1 $ \v -> do
-        guard $ v ^? key "headId" == Just (toJSON headId)
-        guard $ v ^? key "tag" == Just "HeadIsOpen"
-        pure $ v ^? key "utxo"
-      lockedUTxO `shouldBe` Just (toJSON $ UTxO.singleton (scriptTxIn, scriptTxOut))
+      someUTxO <- seedFromFaucet node someExternalVk 10_000_000 (contramap FromFaucet tracer)
+      utxoToCommit <- seedFromFaucet node someExternalVk 5_000_000 (contramap FromFaucet tracer)
+      let someAddress = mkVkAddress networkId someExternalVk
+      let someOutput =
+            TxOut
+              someAddress
+              (lovelaceToValue $ Coin 2_000_000)
+              TxOutDatumNone
+              ReferenceScriptNone
+      buildTransaction networkId nodeSocket someAddress utxoToCommit (fst <$> UTxO.pairs someUTxO) [someOutput] >>= \case
+        Left e -> failure $ show e
+        Right body -> do
+          let unsignedTx = makeSignedTransaction [] body
+          let clientPayload =
+                Aeson.object
+                  [ "blueprintTx" Aeson..= toJSON unsignedTx
+                  , "utxo" Aeson..= toJSON utxoToCommit
+                  ]
+          res <-
+            runReq defaultHttpConfig $
+              req
+                POST
+                (http "127.0.0.1" /: "commit")
+                (ReqBodyJson clientPayload)
+                (Proxy :: Proxy (JsonResponse (DraftCommitTxResponse Tx)))
+                (port $ 4000 + hydraNodeId)
+
+          let DraftCommitTxResponse{commitTx} = responseBody res
```

Comment:
Could: not rely on server-specific data types

This is quite minor, but we could also just assert the response is a serialized transaction:
```suggestion
                (Proxy :: Proxy (JsonResponse Tx))
                (port $ 4000 + hydraNodeId)

          let commitTx = responseBody res
```